libcloudforensics/providers/gcp/internal/compute.py

```diff
@@ -872,19 +877,42 @@ def DetachDisk(self, disk: 'GoogleComputeDisk') -> None:
     response = request.execute()
     self.BlockOperation(response, zone=self.zone)
 
-  def Delete(self, delete_disks: bool = False) -> None:
+  def Delete(
+      self, delete_disks: bool = False, force_delete: bool = False) -> None:
     """Delete an Instance.
 
     Args:
       delete_disks (bool): force delete all attached disks (ignores the 'Keep
-        when instance is deleted' bit).
+          when instance is deleted' bit).
+      force_delete (bool): force delete the instance, even if deletionProtection
+          is set to true.
     """
     disks_to_delete = []
     if delete_disks:
       disks_to_delete = [
           disk['source'].split('/')[-1] for disk in self.GetValue('disks')]
 
     gce_instance_client = self.GceApi().instances()
+
+    if force_delete:
+      if self.deletion_protection:
+        logger.info('Deletion protection detected. Disabling due to '
+                    'force_delete=True')
+        try:
+          request = gce_instance_client.setDeletionProtection(
+              project=self.project_id,
+              zone=self.zone,
+              resource=self.name,
+              deletionProtection=False)
+          response = request.execute()
+        except HttpError as exception:
+          logger.error('Unable to toggle deleteProtection on instance {0:s}: '
+                       '{1:s}'.format(self.name, str(exception)))
+          raise errors.ResourceDeletionError(
+              'Could not delete instance {0:s}: {1!s}'.format(
```

Comment:
Given that we're handling an error related to `setDeletionProtection` and not the deletion itself, can we change this error message to reflect that the error happened while carrying out that operation?